source/server/hot_restarting_parent.cc

```diff
@@ -106,18 +106,25 @@ HotRestartingParent::Internal::getListenSocketsForChild(const HotRestartMessage:
   wrapped_reply.mutable_reply()->mutable_pass_listen_socket()->set_fd(-1);
   Network::Address::InstanceConstSharedPtr addr =
       Network::Utility::resolveUrl(request.pass_listen_socket().address());
+
   for (const auto& listener : server_->listenerManager().listeners()) {
     Network::ListenSocketFactory& socket_factory = listener.get().listenSocketFactory();
     if (*socket_factory.localAddress() == *addr && listener.get().bindToPort()) {
-      // worker_index() will default to 0 if not set which is the behavior before this field
-      // was added. Thus, this should be safe for both roll forward and roll back.
-      if (request.pass_listen_socket().worker_index() < server_->options().concurrency()) {
-        wrapped_reply.mutable_reply()->mutable_pass_listen_socket()->set_fd(
-            socket_factory.getListenSocket(request.pass_listen_socket().worker_index())
-                ->ioHandle()
-                .fdDoNotUse());
+      Network::Socket::Type socket_type =
+          Network::Utility::urlIsTcpScheme(request.pass_listen_socket().address())
```

Comment:
This can be done out of loop, we needn't parse that for each iteration.

Reply:
this will be evaluated only for listeners what match the address, not for every listeners. I will see if it makes sense to take this out of the loop after I update the code to handle the case for unix sockets.